table/iterator.go

```diff
@@ -70,28 +72,65 @@ var (
 	current = 1
 )
 
+func (itr *blockIterator) getKeyAtIndex(idx int) []byte {
+	y.AssertTrue(idx >= 0 && idx < len(itr.entryOffsets))
+
+	idxPos := itr.entryOffsets[idx]
+	var h header
+	idxPos += uint32(h.Decode(itr.data[idxPos:]))
+
+	var idxKey []byte
+	if cap(idxKey) < int(h.plen+h.klen) {
+		sz := int(h.plen) + int(h.klen) // Convert to int before adding to avoid uint16 overflow.
+		idxKey = make([]byte, 2*sz)
```

Comment:
Why do we need to do `2*sz`? wouldn't `idxKey = make([]byte, int(h.plen) + int(h.klen))` be enough?